testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1416,11 +1408,27 @@ public void testAcidUpdateWithSubqueryAssignment()
                 // TODO (https://github.com/trinodb/trino/issues/8268) verifySelectForTrinoAndHive("SELECT * FROM " + tableName, "true", row(1, "AFRICA"), row(2, "AFRICA"));
                 verifySelect("onTrino", onTrino(), "SELECT * FROM " + tableName, "true", row(1, "AFRICA"), row(2, "AFRICA"));
             });
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testAcidUpdateWithCorrelatedSubqueryAssignment()
+    {
+        withTemporaryTable("test_update_subquery", true, false, NONE, tableName -> {
```

Comment:
We could merge it with the previous tests but for HDP 3.1 it throws an IO Excpetion (https://issues.apache.org/jira/browse/HIVE-22318)

Happy to apply a workaround for it 

cc: @losipiuk

Reply:
> We could merge it with the previous tests but for HDP 3.1 it throws an IO Excpetion 

is the test run for any other Hive version than 3?

Reply:
Please merge the test back, so that it's easier to see changes.

see `// TODO (https://github.com/trinodb/trino/issues/8268)` for the problem you're hitting.